File: src/main/java/com/openzoosim/services/users/UserService.java
package com.openzoosim.services.users;

import com.openzoosim.entities.UserEntity;
import com.openzoosim.models.interfaces.IUserService;
import com.openzoosim.repositories.UserRepository;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;

@ApplicationScoped
public class UserService implements IUserService {

    @Inject
    UserRepository _userRepository;

    public UserEntity getUserByID(Long id) {
        return _userRepository.findById(id);
    }

    public UserEntity getUserByEmail(String email) {
        return _userRepository.findByEmail(email.toLowerCase());
    }

    public UserEntity createNewUser(String email, String password, String name) {
        UserEntity user = new UserEntity();
        user.Email = email.toLowerCase();
        user.PasswordHash = password;
        user.Username = name;
        _userRepository.persist(user);
        return user;
    }

    public void updateUser(UserEntity user) {
        _userRepository.persist(user);
    }

}

File: server/services/user.service.go
package services

import (
	"github.com/snowlynxsoftware/ozs-api/server/database/repositories"
)

type UserService struct {
	UserRepository *repositories.UserRepository
	TokenService   *TokenService
}

func NewUserService(userRepository *repositories.UserRepository, tokenService *TokenService) *UserService {
	return &UserService{UserRepository: userRepository, TokenService: tokenService}
}

func (s *UserService) GetUserById(id int) (*repositories.UserEntity, error) {
	return s.UserRepository.GetUserById(id)
}
